### DISCUSSION TITLE
Add MPTCP support for Linux

### DISCUSSION THREAD

**Discussion by vanyingenzi** [Kludex/uvicorn#1655] [Ideas]
MPTCP stands for MultiPath TCP. It is an Open Source protocol that have been standardized and implemented in the Linux Kernel version 5.1.x and above. The kernel support makes it easier to add to applications.

A key thing to understand is that, the protocol is negotiated within the TCP handshake and if any of the parties don't support MPTCP then it will fall back to the normal TCP connection.

The protocol is supported by big companies like Intel, Apple, Redhat, ...

**Comment by euri10** [comment]
Any reason why we should support it ?

**Reply by vanyingenzi** [reply]
MPTCP has proven benefits like : redundancy and aggregated bandwidth.

**Reply by Kludex** [reply]
We need more reasoning here.
